src/main/java/bio/terra/cli/command/app/passthrough/Gcloud.java

```diff
@@ -1,11 +1,44 @@
 package bio.terra.cli.command.app.passthrough;
 
+import bio.terra.cli.businessobject.Context;
+import bio.terra.cli.businessobject.Resource;
+import bio.terra.cli.exception.SystemException;
+import bio.terra.cli.exception.UserActionableException;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
 import picocli.CommandLine.Command;
 
 /** This class corresponds to the second-level "terra gcloud" command. */
-@Command(name = "gcloud", description = "Call gcloud in the Terra workspace.")
+@Command(
+    name = "gcloud",
+    description = "Call gcloud in the Terra workspace.",
+    modelTransformer = Gcloud.SubCmdFilter.class)
 public class Gcloud extends ToolCommand {
 
+  static class SubCmdFilter implements CommandLine.IModelTransformer {
+    public CommandLine.Model.CommandSpec transform(CommandLine.Model.CommandSpec commandSpec) {
+      try {
+        if (!Context.getServer().getCloudBuildEnabled()) {
+          commandSpec.removeSubcommand("--gcs-bucket");
+        }
+      } catch (SystemException e) {
```

Comment:
Why catch and return? What happens if you replace 23-29 with

```
        if (!Context.getServer().getCloudBuildEnabled()) {
          commandSpec.removeSubcommand("--gcs-bucket");
        }
```

Reply:
Hi Melissa, I did this before in the first version. However, at the very very beginning, `Context.getServer()` has no server initialized and would throw an error (See line 175 in Context.java). This try catch is to specially handle the corner case at the initialization stage, and only decide to hide the command once we have the enough (server) information.

Reply:
Thanks. As a general note, for code that's not obvious, it's helpful to add a comment. :)

This is getting complicated. How does this sound:

- Delete lines 21-32
- Print an error if Cloud Build is not enabled and `--gcs-bucket` is set.

That way, if Cloud Build isn't enabled, and user tries to set `--gcs-bucket`, they'll get an error.

Note that noone (no users) is using pools in Broad deployments. It may be years before pools are used.

@yuhuyoyo let me know if you disagree with this suggestion.

Reply:
Discussed offline. We'll keep this.